src/seq/sequence.jl

```diff
@@ -108,6 +108,37 @@ Base.findfirst(seq::Sequence, val) = findnext(seq, val, 1)
 Base.findlast(seq::Sequence, val) = findprev(seq, val, endof(seq))
 
 
+"""
+    gc_content(seq::Sequence)
+
+Calculate GC content of `seq`.
+"""
+function gc_content(seq::Sequence)
+    gc = 0
+
+    if eltype(seq) == DNANucleotide
+        for x in seq
+            if x == DNA_C || x == DNA_G
+                gc += 1
+            end
+        end
+    elseif eltype(seq) == RNANucleotide
```

Comment:
For me it seems this could be written without the `if elseif` statement, by  using Julias dispatch and definining

```
G(::Type{DNANucleotide}) = DNA_G
C(::Type{DNANucleotide}) = DNA_C
G(::Type{RNANucleotide}) = RNA_G
C(::Type{RNANucleotide}) = RNA_C

function gc_content(seq::Sequence)
   gc = 0
   for x in seq
      if x == T(eltype(seq)) || G(eltype(seq))
         gc += 1
      end
   end
   gc
end
```